docs/products/flink/concepts/watermarks.rst

```diff
@@ -0,0 +1,31 @@
+Watermarks
+==========
+
+Another concept closely related to windows and event time in Apache Flink is *watermarks*. Flink uses watermarks as a mechanism to measure progress in event time; they flow as part of the data stream, carrying a timestamp that declares the minimum event time reached in the data stream.
+
+This allows Flink to set points in the stream when all events up to a certain timestamp should have arrived, so that operators can set their internal event time to the value of the watermarks that reach them.
+
+For example, the following diagram shows how watermarks fit into an out-of-order stream, where the events are not ordered by their timestamps:
+
+.. mermaid::
+
+    graph TD;
```

Comment:
The graph now looks VERY long. Not sure there are alternatives...

Reply:
It's a bit of a tricky case - the horizontal orientation means that it's quite small, and reducing the number of items kind of defeats the point of the graph... Would it make sense to leave it out for now?